Add option to use system clipboard for copy and paste.

david-hall opened the pull request:
* Controlled by enable.system.clipboard=true in preferences file.
* Defaults to the original behavior (not using system clipboard)
* No ClipboardController even gets created if the feature is not enabled.

When I realized that you were using a VEF string for implementing your internal clipboard, I thought it might be helpful to quickly capture the selection to the system clipboard instead of just an internal String variable. Then I can paste it to a text editor while I'm working on getting the VEF export to be in canonically sorted order. I found an example of using the system clipboard in java and integrated it along with a quick test case. Hope you like it.

david-hall commented:
I wonder if Travis will have any trouble running the new test case that uses the system clipboard.

vorth commented:
Love it!

Copy-paste has iffy semantics in vZome (paste to where?), but this is certainly a better way to do it, and doesn't change those semantics.

vorth commented:
Damn... doesn't work on the Mac.  There must be some extra integration required.  Still, it does no harm.

david-hall commented:
I assume you saw the required setting in the prefs file to activate this
feature. Bummer. I just assumed it would work on a MAC too since it's a
standard java interface to the clipboard. Maybe I missed something that
Windows doesn't care about, but it works fine here.

On Thu, Aug 11, 2016 at 3:38 AM, Scott Vorthmann <notifications@github.com>
wrote:

> Damn... doesn't work on the Mac. There must be some extra integration
> required. Still, it does no harm.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/vorth/vzome-desktop/pull/33#issuecomment-239092573>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKcyCjn9t3uwh-s2w12kOiWAm39x2g1-ks5qetFlgaJpZM4Jg4PW>
> .
>

vorth commented:
No, too sleepy... didn't notice it.  It works!

vorth commented:
Can you validate the type and content before attempting to parse as VEF on paste?  We don't want horrible runtime errors... something more genteel about "unsupported clipboard content".

david-hall commented:
I thought about that but wanted to see if you even liked the system
clipboard idea before I put that much effort into it. I think it's already
checking for the clipboard to contain text (as opposed to some other format
like an image), but I'm not 100% sure.

What do you think about something as simple as looking for a String
starting with "vZome VEF" from the VEF header and if it's there, then
assume it's good and go ahead and paste it and let the vef parser validate
the rest.

I'll see what I can work out.


On Thu, Aug 11, 2016 at 3:56 AM, Scott Vorthmann <notifications@github.com>
wrote:

> Can you validate the type and content before attempting to parse as VEF on
> paste? We don't want horrible runtime errors... something more genteel
> about "unsupported clipboard content".
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/vorth/vzome-desktop/pull/33#issuecomment-239095727>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKcyCtWTXRSdoL5J87EFeKA9un2Znuxvks5qetXHgaJpZM4Jg4PW>
> .
>

vorth commented:
Yes, that’s about what I had in mind.

Scott

> On Aug 11, 2016, at 1:27 AM, David Hall <notifications@github.com> wrote:
> 
> I thought about that but wanted to see if you even liked the system
> clipboard idea before I put that much effort into it. I think it's already
> checking for the clipboard to contain text (as opposed to some other format
> like an image), but I'm not 100% sure.
> 
> What do you think about something as simple as looking for a String
> starting with "vZome VEF" from the VEF header and if it's there, then
> assume it's good and go ahead and paste it and let the vef parser validate
> the rest.
> 
> I'll see what I can work out.
> 
> 
> On Thu, Aug 11, 2016 at 3:56 AM, Scott Vorthmann <notifications@github.com>
> wrote:
> 
> > Can you validate the type and content before attempting to parse as VEF on
> > paste? We don't want horrible runtime errors... something more genteel
> > about "unsupported clipboard content".
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/vorth/vzome-desktop/pull/33#issuecomment-239095727>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/AKcyCtWTXRSdoL5J87EFeKA9un2Znuxvks5qetXHgaJpZM4Jg4PW>
> > .
> >
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub <https://github.com/vorth/vzome-desktop/pull/33#issuecomment-239100825>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ABgrmJcRYtaiscxUbeVGA-X7fuK2h0LMks5qetzVgaJpZM4Jg4PW>.
>

david-hall commented:
I added the validation to pasteVEF() in vzome-core and added a little
better exception handling in vzome-desktop.

On Thu, Aug 11, 2016 at 10:57 AM, Scott Vorthmann <notifications@github.com>
wrote:

> Yes, that’s about what I had in mind.
>
> Scott
>
> > On Aug 11, 2016, at 1:27 AM, David Hall <notifications@github.com>
> wrote:
> >
> > I thought about that but wanted to see if you even liked the system
> > clipboard idea before I put that much effort into it. I think it's
> already
> > checking for the clipboard to contain text (as opposed to some other
> format
> > like an image), but I'm not 100% sure.
> >
> > What do you think about something as simple as looking for a String
> > starting with "vZome VEF" from the VEF header and if it's there, then
> > assume it's good and go ahead and paste it and let the vef parser
> validate
> > the rest.
> >
> > I'll see what I can work out.
> >
> >
> > On Thu, Aug 11, 2016 at 3:56 AM, Scott Vorthmann <
> notifications@github.com>
> > wrote:
> >
> > > Can you validate the type and content before attempting to parse as
> VEF on
> > > paste? We don't want horrible runtime errors... something more genteel
> > > about "unsupported clipboard content".
> > >
> > > —
> > > You are receiving this because you authored the thread.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/vorth/vzome-desktop/pull/33#issuecomment-239095727
> >,
> > > or mute the thread
> > > <https://github.com/notifications/unsubscribe-auth/
> AKcyCtWTXRSdoL5J87EFeKA9un2Znuxvks5qetXHgaJpZM4Jg4PW>
> > > .
> > >
> > —
> > You are receiving this because you modified the open/close state.
> > Reply to this email directly, view it on GitHub <
> https://github.com/vorth/vzome-desktop/pull/33#issuecomment-239100825>,
> or mute the thread <https://github.com/notifications/unsubscribe-
> auth/ABgrmJcRYtaiscxUbeVGA-X7fuK2h0LMks5qetzVgaJpZM4Jg4PW>.
>
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/vorth/vzome-desktop/pull/33#issuecomment-239186912>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKcyChUtcp-MQirbUsqnJk_xBqNg-wxkks5qezhKgaJpZM4Jg4PW>
> .
>
